### PR TITLE
[luci] Migrate helperPads to ShapeInferenceHelper

### PR DESCRIPTION
**Pull request by icodo98**
This PR migrates heplerPads.h to CircleShapeInferenceHelper.h and CircleShapeInferenceHelper.cpp.

ONE-DCO-1.0-Signed-off-by: JuYoung Lee rsb98759@gmail.com

- background : https://github.com/Samsung/ONE/pull/13780#discussion_r1735549476

### CODE REVIEW

**Review by zetwhite — APPROVED** [review]
LGTM

File: compiler/luci/service/src/CircleShapeInferenceHelper.cpp [hunk 1]
```diff
@@ -16,6 +16,10 @@
 
 #include "CircleShapeInferenceHelper.h"
 
+#include "Check.h"
+
+#include <limits>
+
 #include <oops/InternalExn.h>
```
Comment (seanshpark) [inline compiler/luci/service/src/CircleShapeInferenceHelper.cpp:23]:
```suggestion
#include <oops/InternalExn.h>

#include <limits>
```

File: compiler/luci/service/src/Nodes/CirclePad.cpp [hunk 2]
```diff
@@ -35,7 +34,8 @@ loco::TensorShape Algorithm::visit(const luci::CirclePad *node)
 {
   // TODO support non-const case
   auto paddings = loco::must_cast<luci::CircleConst *>(node->paddings());
-  return use_paddings(node, paddings);
+  auto input_shape = circle_shape(loco::must_cast<const luci::CircleNode *>(node->input()));
```
Comment (seanshpark) [inline compiler/luci/service/src/Nodes/CirclePad.cpp:37]:
```suggestion
  auto input = loco::must_cast<const luci::CircleNode *>(node->input());
  auto input_shape = circle_shape(input);
```

Comment (icodo98) [reply compiler/luci/service/src/Nodes/CirclePad.cpp:37]: How about
```suggestion
  auto circle_input = loco::must_cast<const luci::CircleNode *>(node->input());
  auto input_shape = circle_shape(input);
```
?
Just for more clarity.

**Review by shs-park — APPROVED** [review]
LGTM!
=)

**Review by llFreetimell — APPROVED** [review]
LGTM :)

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by zetwhite — APPROVED** [review]

**Pull request merged**